src/figuresClasses.ts

```diff
@@ -1,11 +1,79 @@
-export interface Figure {}
+export interface Figure {
+  shape: string;
+  color: string;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: string = 'triangle';
+
+  constructor(
+    public color: string,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const sizes: number[] = [a, b, c];
+    const maxSize: number | undefined = sizes
+      .sort((aSide: number, bSide: number) => {
+        return aSide - bSide;
+      })
+      .pop();
+    const lowerSidesSum = sizes.reduce((acc, side) => acc + side, 0);
+
+    if (a <= 0 || b <= 0 || c <= 0 || (maxSize && lowerSidesSum <= maxSize)) {
+      throw new Error('Triangle side cannot be <= 0');
+    }
+  }
+
+  getArea(): number {
+    const semiperimeter: number = (this.a + this.b + this.c) / 2;
+    const area: number = Math.sqrt(
+      semiperimeter *
+        (semiperimeter - this.a) *
+        (semiperimeter - this.b) *
+        (semiperimeter - this.c),
+    );
+
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Circle implements Figure {
+  shape: string = 'circle';
+
+  constructor(
+    public color: string,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('Circle radius cannot be <= 0');
+    }
+  }
 
-export class Triangle implements Figure {}
+  getArea(): number {
+    return Math.floor(Math.PI * Math.pow(this.radius, 2) * 100) / 100;
+  }
+}
+
+export class Rectangle implements Figure {
+  shape: string = 'rectangle';
 
-export class Circle implements Figure {}
+  constructor(
+    public color: string,
+    public width: number,
+    public height: number,
+  ) {
+    if (width <= 0 || height <= 0) {
+      throw new Error('Rectangle side cannot be <= 0');
+    }
+  }
 
-export class Rectangle implements Figure {}
+  getArea(): number {
```

Comment:
The 'getArea' method should not accept any parameters. It should use the object's own fields to calculate the area.